.github/workflows/test.yml

```diff
@@ -0,0 +1,98 @@
+name: test
+on: pull_request
+jobs:
+  unit:
+    runs-on: ubuntu-latest
+    env:
+      DATABASE_URL: postgresql://hatchet:hatchet@127.0.0.1:5431/hatchet
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v2
+        with:
+          version: '23.x'
+
+      - name: Install Task
+        uses: arduino/setup-task@v1
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
+
+      - name: Cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache
+          restore-keys: ${{ runner.os }}-go-
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+
+      - name: Compose
+        run: docker compose up -d
```

Comment:
`unit` tests were not supposed to require compose, however the sqlc generation depends on prisma diff for the sql schema and that requires a live DB. Can be refactored later